lib/Conf.mli

```diff
@@ -100,17 +101,15 @@ type action =
       (** Check whether the input files already are formatted. *)
   | Print_config of t  (** Print the configuration and exit. *)
 
-val action : unit -> action
-(** Formatting action: input type and source, and output destination. *)
-
-val debug : bool
-(** Generate debugging output if true. *)
+type opts =
+  { debug: bool  (** Generate debugging output if true. *)
+  ; check: bool  (** Check whether the input files already are formatted. *)
+  ; margin_check: bool
+        (** Check whether the formatted output exceeds the margin. *) }
+(** Options changing the tool's behavior *)
 
-val check : bool
-(** Check whether the input files already are formatted. *)
-
-val margin_check : bool
-(** Check whether the formatted output exceeds the margin. *)
+val action : unit -> action * opts
```

Comment:
Rather than returning a tuple, what do you think of adding the booleans to the individual actions? for example, if `action` is `In_out`, `check` is not used.

Reply:
`check` is not necessary and *must* not be there !
The `opts` record may contain more fields in the future (taken from `Conf.t`) and makes sense on its own. I don't agree to inline it into actions.